src/android/toga_android/paths.py

```diff
@@ -13,18 +13,13 @@ class Paths:
     def __context(self):
         return App.app._impl.native.getApplicationContext()
 
+    def __init__(self):
+        # On Android, __main__ only exists during app startup, so cache its location now.
+        self._app = Path(sys.modules["__main__"].__file__).parent
+
     @property
     def app(self):
-        try:
-            return Path(sys.modules["__main__"].__file__).parent
-        except KeyError:
-            # If we're running in test conditions,
-            # there is no __main__ module.
-            return Path.cwd()
```

Comment:
`Path.cwd` is useless on Android, because it usually returns the root directory, which isn't even readable on most devices.